Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -101,12 +101,34 @@ class APIClient {
         recoveryQueue.addOperation(requestOperation)
     }
 
+    /// Performs a network request and retries in case of network failures. The network operation
+    /// won't be managed by the APIClient instance. Instead it will be added on the provided operationQueue.
+    ///
+    /// - Parameters:
+    ///   - endpoint: The `Endpoint` used to create the network request.
+    ///   - operationQueue: The queue that will be responsible for executing the network operation
+    ///   - completion: Called when the networking request is finished.
+    func unmanagedRequest<Response: Decodable>(
+        endpoint: Endpoint<Response>,
+        operationQueue: OperationQueue,
```

Comment:
Given the usage, I think we can safely ignore this parameter and use .main

Reply:
I would also rename it to `priorityRequest`

Reply:
I'm a fan of `priorityRequest`. We aren't really applying any priority logic, we are just enqueuing in another queue. Unmanaged seems to be closer in accordance to what it does. Additionally it's also aligned with the Swift's Unmanaged API for manual (memory) management